src/settings/RepositoryTableModel.cpp

```diff
@@ -0,0 +1,151 @@
+#include "RepositoryTableModel.h"
+#include <QSet>
+
+RepositoryTableModel::RepositoryTableModel(QObject * parent)
+: QAbstractTableModel(parent) {
+    load({});
+}
+
+QList<QStringList> RepositoryTableModel::dump() const {
+    QList<QStringList> out;
+    for (auto &&repo : repositories_) {
+        if (repo.isDefault) continue;
+        out << QStringList{repo.name, repo.url};
+    }
+    return out;
+}
+
+void RepositoryTableModel::load(QList<QStringList> data) {
```

Comment:
Somewhat, but that's the (small) price to pay for storing it as a StringList in the settings, and making Settings the single source of truth for now. This also allows the internal data model for RepositoryTableManager and RepoManager to be different, which makes those things a little more self contained.

Reply:
Oook, I'm getting a bit confused here. So. When loading the model, `RepoManager` inserts the default repository. All other repository insertions are managed by `RepoTableModel`?
`RepoManager` just serves as a container ObjectEssentially to feed `Repositories` inside `ModelManager`. Honestly, we can just remove most of it and rely on `RepositoryTableModel`.  What do you think?

Reply:
I did some playing around:
ModelManager and Mainwindow can both access `settings_->externalRepos()`. If we make the default contain the default repository (eg settings_->externalRepos() becomes settings_->repos()) and prevent it from being deleteable and make everything read it directly out of settings, we don't need `inventory/RepoManager*`. This is a small refactoring and I'm happy to keep it out of this PR, so t we don't block anything else and do it later in separate PR. If you agree with me, I'm happy to have this merged now.

Reply:
You're right, RepoManager is just a substructure of ModelManager at this point. RepoManager exists to add that default entry. I suppose that is something Settings can do very well on its own as well. I'll change that, it makes things simpler :)

One of the reasons for the current set-up, in which RepositoryTableModel is not the single source of truth, is to allow you to edit the table in the settings panel, but then press "Cancel" and not have those changes take effect. It forces Settings to be the source of truth, and allows us to control when that one is actually updated[^1].

[^1]: The translation settings window is a bit of an inconsistent thing right now anyway: Changes in the model table do have immediate effect, but all the other panes in that window do not.

I'm also not really a fan of how the ModelManager interface looks right now. It does too many things… and I wouldn't mind if I can move the bits that provide the data for rendering the table and the dropdown menu to their own classes. I don't know how I could make Settings also the single source here, since adding/removing models is much more involved than updating a value in a config file.